conf/mpidistutils.py

```diff
@@ -908,13 +908,44 @@ def configure_mpi(ext, config_cmd):
       return 0;
     }
     """)
-    errmsg = "Cannot {} MPI programs. Check your configuration!!!"
+    errmsg = [
+        "Cannot {} MPI programs. Check your configuration!!!",
+        "Installing mpi4py requires a working MPI implementation.",
+    ]
+    if sys.platform == 'linux':
+        errmsg += [
+            "If you are running on a supercomputer or cluster, check with",
+            "the system administrator or refer to the system user guide.",
+            "Otherwise, if you are running on a laptop or desktop computer,",
+            "your may be missing the MPICH or Open MPI development package:",
+            "* On conda environments with conda-forge packages:",
+            "  $ conda install -c conda-forge mpich-mpicc   # for MPICH",
+            "  $ conda install -c conda-forge openmpi-mpicc # for Open MPI",
+            "* On Fedora/RHEL systems:",
+            "  $ sudo dnf install mpich-devel     # for MPICH",
+            "  $ sudo dnf install openmpi-devel   # for Open MPI",
+            "* On Debian/Ubuntu systems:",
+            "  $ sudo apt install libmpich-dev    # for MPICH",
+            "  $ sudo apt install libopenmpi-dev  # for Open MPI",
+        ]
+    if sys.platform == 'darwin':
+        errmsg += [
+            "Install MPICH or Open MPI with Homebrew or MacPorts:"
+            "  $ brew install mpich|openmpi  # Homebrew",
+            "  $ port install mpich|openmpi  # MacPorts",
+        ]
+    if sys.platform == 'win32':
+        errmsg += [
+            "Please install *Intel MPI* or *Microsoft MPI*."
+        ]
```

Comment:
I think msmpi is also available from conda-forge, not sure about intel mpi, though

Reply:
Yes, not Intel MPI is also available. But think we are making a mistake mentioning conda-forge for the `pip install case`. An experienced user know what he/she is doing, a novice/beginner should probably stay away from mixing pip packages with conda packages.

Reply:
Wouldn't this be the same case for brew/port?

Reply:
Well, yes, you have a point. Now that I've spent so much time in it, let me do the full think at least for Linux and macOS. As usual, Windows may require more careful considerations I'm not willing to make for free.